tests/python/pants_test/engine/test_engine.py

```diff
@@ -119,16 +119,16 @@ def test_include_trace_error_raises_error_with_trace(self):
 
     self.assert_equal_with_printing(dedent('''
       1 Exception encountered:
-      Computing Select(<pants_test.engine.test_engine.B object at 0xEEEEEEEEE>, A)
-        Computing Task(nested_raise(), <pants_test.engine.test_engine.B object at 0xEEEEEEEEE>, A, true)
+      Computing Select(<tests.python.pants_test.engine.test_engine.B object at 0xEEEEEEEEE>, A)
```

Comment:
This is a potentially problematic step backward. It means that we're not correctly detecting the name of the module relative to the sourceroot (although only for tests, perhaps? that's a bit odd). I suspect that it would mean that imports would break in some situations.

Reply:
>  I suspect that it would mean that imports would break in some situations.

We support both types of imports now: `tests.python.pants_test.engine.scheduler_test_base` and `pants_test.engine.scheduler_test_base`, whereas V1 only supports the latter.

> It means that we're not correctly detecting the name of the module relative to the sourceroot (although only for tests, perhaps? that's a bit odd).

Sort of. It means that Python uses the full module name `tests.python.pants_test.engine.scheduler_test_base`, and we still allow using the source-root stripped version `pants_test.engine.scheduler_test_base`. Both are valid ways to refer to the Python modules; the former is more just more explicit, so Python uses it when printing things like `repr`.

Regardless, don't see a way to do this without stripping the prefix from files. I'm still unclear how V1 did this all, even after reading the V1 Pytest code dozens of times the past year :/

Reply:
The answer is probably related to the pytest module that (I had forgotten that) we install: https://github.com/pantsbuild/pants/blob/master/src/python/pants/backend/python/tasks/pytest/plugin.py ...

Reply:
> 

See here:
+ pytest plugin code: https://github.com/pantsbuild/pants/blob/a996b570c0ad637e60ebfb62a66182dbf72d139a/src/python/pants/backend/python/tasks/pytest/plugin.py
+ configuration of the plugin:
https://github.com/pantsbuild/pants/blob/cdb220720c5cdf3ab2903aedde0dac0778c3bc3a/src/python/pants/backend/python/tasks/pytest_run.py#L368-L383

Reply:
I figured out the line that I was missing for the life of me! https://github.com/pantsbuild/pants/pull/8180/files#diff-5f2c007abfa9fbf0323826392a6374c7R101

And, as a result, I think I understand a bit better now how this plugin comes into play.